tests/unit/store/tenant.spec.js

```diff
@@ -0,0 +1,369 @@
+        expect(store.getters['tenant']).toEqual({
+          ...loggedinTenantState().tenant,
+          settings: { agreedToLicense: true }
+        })
+      })
+      it('should throw an error message if no settings are passed', () => {
+        expect(() => store.commit('updateTenantSettings')).toThrow(
+          'passed invalid or empty settings object'
+        )
+      })
+    })
+  })
+
+  describe('actions', () => {
+    describe('getTenants - no query error', () => {
+      let store
+
+      beforeEach(() => {
+        mockerror = false
+        store = new Vuex.Store({
+          state: initialTenantState(),
+          getters: tenant.getters,
+          mutations: tenant.mutations,
+          actions: tenant.actions
+        })
+      })
+
+      it('should set tenants', async () => {
+        await store.dispatch('getTenants')
+        expect(prefectTenants).toHaveBeenCalled()
+      })
+
+      it('should set tenants', async () => {
+        const tenantsArray = [{ name: 'boo', id: '12345' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        await store.dispatch('getTenants')
+        expect(store.getters.tenants).toEqual(tenantsArray)
+      })
+
+      it('should return tenants', async () => {
+        const tenantsArray = [{ name: 'boo', id: '12345' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        const returnedTenants = await store.dispatch('getTenants')
+        expect(returnedTenants).toEqual([{ name: 'boo', id: '12345' }])
+      })
+    })
+
+    describe('setCurrentTenant -server', () => {
+      let store
+
+      beforeEach(() => {
+        store = new Vuex.Store({
+          state: initialTenantState(),
+          getters: tenant.getters,
+          mutations: tenant.mutations,
+          actions: tenant.actions
+        })
+      })
+
+      it('should throw an error if no slug is provided', () => {
+        expect(store.dispatch('setCurrentTenant', false)).rejects.toThrow(
+          'No slug was provided when trying to set the current tenant'
+        )
+      })
+
+      it('should check if tenant is set and getTenant if not set', async () => {
+        expect(store.getters.tenant.slug).toBe(null)
+        const tenantsArray = [{ name: 'boo', id: '12345', slug: 'team2' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        await store.dispatch('setCurrentTenant', 'team2')
+        expect(store.getters.tenant.slug).toEqual('team2')
+      })
+
+      it('should throw an error if the requested tenant does not exist', async () => {
+        expect(store.getters.tenant.slug).toBe(null)
+        const tenantsArray = [{ name: 'boo', id: '12345', slug: 'team2' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        expect(store.dispatch('setCurrentTenant', 'team3')).rejects.toThrow(
+          "Error: Unable to set current tenant: tenant doesn't exist"
+        )
+      })
+
+      it('should set the tenant role to Tenant Admin', async () => {
```

Comment:
```suggestion
      it('should set the tenant role to TENANT_ADMIN', async () => {
```

Reply:
Can you add a comment describing why this is the case? It'll be useful when debugging this for someone unfamiliar with the differences between Server and Cloud.